include/class_loader/class_loader.h

```diff
@@ -100,54 +108,62 @@ class ClassLoader
   }
 
   /**
-   * @brief  Generates an instance of loadable classes (i.e. class_loader). It is not necessary for the user to call loadLibrary() as it will be invoked automatically if the library is not yet loaded (which typically happens when in "On Demand Load/Unload" mode).
+   * @brief  Generates an instance of loadable classes (i.e. class_loader).
+   *
+   * It is not necessary for the user to call loadLibrary() as it will be invoked automatically
+   * if the library is not yet loaded (which typically happens when in "On Demand Load/Unload" mode).
+   *
    * @param  derived_class_name The name of the class we want to create (@see getAvailableClasses())
    * @return A std::shared_ptr<Base> to newly created plugin object
    */
   template <class Base>
   std::shared_ptr<Base> createInstance(const std::string & derived_class_name)
   {
-    if (ClassLoader::hasUnmanagedInstanceBeenCreated() && isOnDemandLoadUnloadEnabled()) {
-      CONSOLE_BRIDGE_logInform(
-        "class_loader::ClassLoader: "
-        "An attempt is being made to create a managed plugin instance (i.e. std::shared_ptr), "
-        "however an unmanaged instance was created within this process address space. "
-        "This means libraries for the managed instances will not be shutdown automatically on "
-        "final plugin destruction if on demand (lazy) loading/unloading mode is used.");
-      }
-
-    if (!isLibraryLoaded()) {
-      loadLibrary();
-    }
-
-    Base * obj = class_loader::impl::createInstance<Base>(derived_class_name, this);
-    assert(obj != nullptr);  // Unreachable assertion if createInstance() throws on failure
-
-    std::lock_guard<std::recursive_mutex> lock(plugin_ref_count_mutex_);
-    ++plugin_ref_count_;
+    return std::shared_ptr<Base>(
+      createRawInstance<Base>(derived_class_name, true),
+      std::bind(&ClassLoader::onPluginDeletion<Base>, this, std::placeholders::_1)
+    );
+  }
 
-    std::shared_ptr<Base> smart_obj(
-      obj, std::bind(&class_loader::ClassLoader::onPluginDeletion<Base>, this, std::placeholders::_1));
-    return smart_obj;
+#if __cplusplus >= 201103L
+  /// Generates an instance of loadable classes (i.e. class_loader).
+  /**
+   * It is not necessary for the user to call loadLibrary() as it will be
+   * invoked automatically.
+   * If the library is not yet loaded (which typically happens when in
```

Comment:
Nit: I think this is the continuation of the previous sentence and not a new one (same below)